Processing/src/cz/fidentis/processing/comparison/surfaceComparison/SurfaceComparisonProcessing.java

```diff
@@ -290,115 +290,233 @@ public void processOneToOne(KdTree mainF, Model compF, int numberOfIterations, b
+        
+        List<Future<List<Vector3f>>> translations = new ArrayList<>(compFs.size());
+        
+        //change to directory of current iteration
+        String currentTMP = FileUtils.instance().getTempDirectoryPath() + File.separator +  tmpLocFile.getPath() + File.separator + 
+                 tmpModuleFile.getName() + "_" + currentBatchIteration + "_";
+        
+        //get list with empty vectors of size of template to add translations to
+        List<Vector3f> trans = ListUtils.instance().populateVectorList(template.getVerts().size());  
+        
         ExecutorService executor;
-
-        //start computing algorithm
-        for (int i = 0; i < numberOfBatchIterations; i++) {
-            executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());        //gets pool of threads based on number of available processors
-            trans.clear();
-            list.clear();
-
-            for (Vector3f o : template.getVerts()) {
-                trans.add(new Vector3f());
-            }
-
-            //aligns all faces performed parallely based on size of execturo pool
-            for (int j = 0; j < compFs.size(); j++) {
-                Model currentModel = ml.loadModel(new File(currentTMP + j + File.separator + tmpModuleFile.getName() + "_" + i + "_" + j + ".obj"), Boolean.FALSE, false);
-                List<Vector3f> samples = getUndersampledMesh(m, t, value, currentModel);
-
-                Future<List<Vector3f>> future = executor.submit(new BatchProcessingCallable(currentModel, samples, template, templateTree,
-                        error, numberOfICPiteration, scale, tmpLocFile, j, i, Boolean.TRUE, metric, data));
-                list.add(future);
-            }
-
-            currentTMP = FileUtils.instance().getTempDirectoryPath() + File.separator +  tmpLoc + "_" + (i + 1) + "_";
-
-            //computes translation vector for each vertex of template face
-            for (Future<List<Vector3f>> list1 : list) {
-                addTranslationToModel(trans, list1);
-            }
-
-            k = ProgressHandleFactory.createHandle("Computing Average Face.");
-            k.start();
-
-            try {
-                //average translation vector for each vertex by number of faces aligned to template and apply it, creating new average face
-                computeMeanTranslationToModel(template, trans, templateSize, compFs.size());
-
-                //creates new kd-tree for just created average face
-                templateTree = new KdTreeIndexed(template.getVerts());
-                k.finish();
-            } catch (Exception ex) {
-                Exceptions.printStackTrace(ex);
-                System.err.print(ex);
-            }finally{
-                k.finish();
-            }
-            executor.shutdown();
+        executor = Executors.newFixedThreadPool(1);

```

Comment:
nechapem moc tento kod ani neviem presne ako funguje ten executor ale len tak rozmyslam - ma zmysel vytvarat si thread pool s jednym threadom? Nie je jednoduchsie to spravit bez toho? (kvoli dodatocnej rezii s vytvaranim threadu atd)